crates/wit-parser/src/ast.rs

```diff
@@ -24,12 +24,14 @@ impl<'a> Ast<'a> {
     pub fn parse(lexer: &mut Tokenizer<'a>) -> Result<Self> {
         let mut items = Vec::new();
         let mut package_id = None;
+        let mut docs = parse_docs(lexer)?;
         if lexer.eat(Token::Package)? {
-            package_id = Some(PackageName::parse(lexer)?);
+            let docs = std::mem::take(&mut docs);
```

Comment:
I think this may mean:

```
/// this works
package foo:bar

/// this gets lost
interface foo {}
```

perhaps?

Reply:
Ugh yes. I rewrote this logic twice already!